### DISCUSSION TITLE
Vectara integration question...

### DISCUSSION THREAD

**Discussion by ofermend** [microsoft/semantic-kernel#5933] [General]
What is the best way to add Vectara as a memoryStore?
Being RAG-as-a-service, the Vectara the interface of the query API is at the query level, and embedding occurs inside the Vectara backend. I see that the interface of MemoryStoreBase assumes it has to be at the embedding level.
Any suggestion for what the best way to integrate is?

**Comment by sophialagerkranspandey** [comment]
We are currently working on updating our memory extractions. In the meantime, feel free to build something using this example: https://github.com/microsoft/semantic-kernel/blob/f66a30bbd9bee1715062b07cfad537fc10f54aee/dotnet/samples/Concepts/Search/MyAzureAISearchPlugin.cs
